lib/modules/manager/npm/update/locked-dependency/common/parent-version.ts

```diff
@@ -1,10 +1,26 @@
 import { logger } from '../../../../../../logger';
 import {
   GetPkgReleasesConfig,
+  ReleaseResult,
   getPkgReleases,
 } from '../../../../../datasource';
 import { api as semver } from '../../../../../versioning/npm';
 
+const pkgCache = new Map<string, Promise<ReleaseResult | null>>();
+
+function getPkgReleasesCached(depName: string): Promise<ReleaseResult | null> {
```

Comment:
I was surprised that this caching didn't occur in the datasource index layer *or* the http layer. But I plan to investigate/improve datasource concurrency and caching at a later time